Testing/ORKTest/ORKTest/MainViewController.m

```diff
@@ -2244,6 +2248,366 @@ - (void)optionalFormButtonTapped:(id)sender {
+    }
+    
+    {
+        // If we arrived at question_03_fail then fall through to question 4
+        ORKDirectStepNavigationRule *directRule = nil;
+        directRule = [[ORKDirectStepNavigationRule alloc] initWithDestinationStepIdentifier:@"question_04"];
+        [task setNavigationRule:directRule forTriggerStepIdentifier:@"question_03_fail"];
+    }
+    
+    
+    {
+        // If we answer 'Yes' to Question 4, then proceed to the pass screen
+        ORKResultSelector *resultSelector = [ORKResultSelector selectorWithResultIdentifier:@"question_04"];
+        NSPredicate *predicateQuestion = [ORKResultPredicate predicateForChoiceQuestionResultWithResultSelector:resultSelector matchingPattern:@"Cat*"];
+        
+        ORKPredicateStepNavigationRule *predicateRule = [[ORKPredicateStepNavigationRule alloc] initWithResultPredicates:@[predicateQuestion]
+                                                                                              destinationStepIdentifiers:@[@"question_04_pass"]];
+        [task setNavigationRule:predicateRule forTriggerStepIdentifier:@"question_04"];
+    }
+    
+    {
+        // If we arrived at question_04_fail then fall through to question 5
+        ORKDirectStepNavigationRule *directRule = nil;
+        directRule = [[ORKDirectStepNavigationRule alloc] initWithDestinationStepIdentifier:@"question_05"];
+        [task setNavigationRule:directRule forTriggerStepIdentifier:@"question_04_fail"];
+    }
+
+    
+    {
+        // If we answer 'Dog' to Question 5, then proceed to the pass screen
+        ORKResultSelector *resultSelector = [ORKResultSelector selectorWithResultIdentifier:@"question_05"];
+        NSPredicate *predicateQuestion = [ORKResultPredicate predicateForTextQuestionResultWithResultSelector:resultSelector expectedString:@"Dog"];
+        
+        ORKPredicateStepNavigationRule *predicateRule = [[ORKPredicateStepNavigationRule alloc] initWithResultPredicates:@[predicateQuestion]
+                                                                                              destinationStepIdentifiers:@[@"question_05_pass"]];
+        [task setNavigationRule:predicateRule forTriggerStepIdentifier:@"question_05"];
+    }
+    
+    {
+        // If we arrived at question_05_fail then fall through to question 6
+        ORKDirectStepNavigationRule *directRule = nil;
+        directRule = [[ORKDirectStepNavigationRule alloc] initWithDestinationStepIdentifier:@"question_07"];
+        [task setNavigationRule:directRule forTriggerStepIdentifier:@"question_05_fail"];
+    }
+    
+    
+/*    {
+        // If we answer '*og' to Question 6, then proceed to the pass screen
+        ORKResultSelector *resultSelector = [ORKResultSelector selectorWithResultIdentifier:@"question_06"];
+        NSPredicate *predicateQuestion = [ORKResultPredicate predicateForTextQuestionResultWithResultSelector:resultSelector matchingPattern:@"*og"];
```

Comment:
Same as above, the correct pattern is `.*og`.

`*og` is an invalid regular expression because there no character preceding the `*` symbol. :)